assets/dev/js/frontend/utils/assets-loader.js

```diff
@@ -0,0 +1,66 @@
+export default class AssetsLoader {
+	getScriptElement( src ) {
+		const scriptElement = document.createElement( 'script' );
+
+		scriptElement.src = src;
+
+		return scriptElement;
+	}
+
+	getStyleElement( src ) {
+		const styleElement = document.createElement( 'link' );
+
+		styleElement.rel = 'stylesheet';
+		styleElement.href = src;
+
+		return styleElement;
+	}
+
+	add( assetData ) {
+		const { type, key, src, parent } = assetData;
+
+		if ( ! this.constructor.assets[ type ][ key ] ) {
+			this.constructor.assets[ type ][ key ] = {
+				src,
+				parent,
+			};
+		}
+	}
```

Comment:
Instead of an `add` method, allow the `load` method to receive a 3rd optional `assetData` parameter

Reply:
As agreed, for now removing the `add` method,